setup.py

```diff
@@ -152,7 +154,8 @@
         ],
         flask=[
             'flask~=1.1;python_version~="3.5.0"',
-            'flask<3;python_version>"3.5"',
+            'flask<3;python_version>"3.5" and python_version<"3.12"',
+            'flask>3,<4;python_version>="3.12"',
```

Comment:
What prevents Flask 2.x from working on python 3.12? Same question applies to all the other updated deps.

Reply:
@goulinkh Good question which I have not thought about. Any reasons? We can anytime remove the strict restriction if things work with older versions, too.

Reply:
Sorry for the delayed response.

I think I misunderstood the version requirement role, If understand correctly this should enforce a recent version but instead have a wide range of available version with the only limitation the minimal working python version.
I've done some research and have done a new version of this, let me know if this new version is valid for `Flask` and `Werkzeug` so I can continue with this logic on the rest of the changed requirements:
```
	Werkzeug~=1.0;python_version~="3.5.0"
	Werkzeug<2.1;python_version>="3.6" and python_version<"3.7"
	Werkzeug<2.3;python_version>="3.7" and python_version<"3.8"
	Werkzeug<3.1;python_version>="3.8"
	
        flask~=1.1;python_version~="3.5.0"
	flask<2.1;python_version>=3.6 and python_version<"3.7"
	flask<2.3;python_version>=3.7 and python_version<"3.8"
	flask<3.1;python_version>=3.8
```

Reply:
This has been addressed here: https://github.com/canonical-ols/talisker/pull/608